micro-rdk/src/common/config_monitor.rs

```diff
@@ -43,7 +69,13 @@ impl<'a> PeriodicAppClientTask for ConfigMonitor<'a> {
                 app_client.clone().get_config(None).await
             {
                 if self.curr_config != *new_config {
-                    self.restart()
+                    if let Err(e) = self.storage.reset_robot_configuration() {
+                        log::warn!(
+                            "Failed to reset robot config after new config detected: {}",
+                            e
+                        );
+                    }
+                    self.restart();
```

Comment:
Does it make more sense to restart if the config cache wasn't reset or to not reset? I'm leaning towards reset because it could fix the reason why the cache failed to erase.

Reply:
I think I agree. If you fail to clear the cache, if you restart you will run the stale config, and if you don't restart you will continue to run the stale config.

Reply:
But if NVS writes aren't working, your best hope is that a reboot will help.

Reply:
Meanwhile, we spent a lot of time talking about why this clears the cache rather than writing the configuration to it. I think a fairly detailed comment is warranted explaining why resetting the cache and rebooting is our current best strategy.

Reply:
So we prefer a (possible reboot loop on an NVS write fail) over a (basically permanently stale config cache assuming NVS doesn't fix itself)?

Reply:
I just realized when writing my comment the alternative, which is nvs fail -> reboot -> nvs fail -> reboot ... and the config cache never changing. While rebooting may fix the issue, maybe a stale config is preferred over a reboot loop?

Reply:
That's a good point. I think I agree then: if you can't clear NVS you are better off continuing with your existing config than entering a boot loop driven by continually detecting a new config. I also expect that being unable to write to NVS probably represents a badly degraded device or runtime, and our options are limited there. It is a fussy edge case though, and I'd be interested to hear if @npmenard has thoughts on the best outcome here.